src/WebJobs.Script.WebHost/Models/HostAssignmentContext.cs

```diff
@@ -24,29 +24,55 @@ public class HostAssignmentContext
         [JsonProperty("MSISpecializationPayload")]
         public MSIContext MSIContext { get; set; }
 
-        public string ZipUrl
+        public HostAssignmentZipUrl ZipUrl
         {
             get
             {
-                if (Environment.ContainsKey(EnvironmentSettingNames.AzureWebsiteRunFromPackage))
-                {
-                    return Environment[EnvironmentSettingNames.AzureWebsiteRunFromPackage];
-                }
-                else if (Environment.ContainsKey(EnvironmentSettingNames.AzureWebsiteAltZipDeployment))
-                {
-                    return Environment[EnvironmentSettingNames.AzureWebsiteAltZipDeployment];
-                }
-                else if (Environment.ContainsKey(EnvironmentSettingNames.AzureWebsiteZipDeployment))
+                string zipUrlEnvVar = GetDefinedZipUrlEnvVar();
+
+                if (!string.IsNullOrEmpty(zipUrlEnvVar))
```

Comment:
Looks like we can just change 

return Environment[EnvironmentSettingNames.AzureWebsiteRunFromPackage];

to

return new HostAssignmentZipUrl(EnvironmentSettingNames.AzureWebsiteRunFromPackage, Environment[EnvironmentSettingNames.AzureWebsiteRunFromPackage])

and that already takes care of precendence also. we dont need GetDefinedZipUrlEnvVar